test/system/trader.js

```diff
@@ -77,11 +77,28 @@ class Trader {
     return new Promise((resolve) => {
       if(side === 'B') {
         this.getLastBuyPriceAndQuantity(price, quantity, side);
-        resolve(this.createOrder(price, side, quantity, options));
+        resolve(this.createOrder(price, side, quantity, 'L', options));
       }
       else {
         this.getLastSellPriceAndQuantity(price, quantity, side);
-        resolve(this.createOrder(price, side, quantity, options));
+        resolve(this.createOrder(price, side, quantity, 'L', options));
+      }
+    });
+  }
+
+  placeLimitOrderSpread(side, price, quantity, count, margin) {
+    return new Promise((resolve) => {
+      for (let i=1; i<=count; i++) {
+        if(side === 'B') {
+          this.getLastBuyPriceAndQuantity(price, quantity, side);
+          price = price - margin;
+          resolve(this.createOrder(price, side, quantity, 'L'));
```

Comment:
The new Promise will resolve as soon as the first call to resolve() is made. You need to wait for all order creations are complete. Something like this...

```javascript
const promises = [];
for (let i = 1; i <= count; i++) {
  promises.push(
    this.createOrder(price + i * margin, 'S', quantity, 'L')
  ); 
  promises.push(
    this.createOrder(price - i * margin, 'B', quantity, 'L')
  ); 
}

return Promish.all(promises);
```